sci-libs/composable-kernel/composable-kernel-5.7.1-r1.ebuild

```diff
@@ -0,0 +1,81 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+ROCM_VERSION=${PV}
+
+inherit cmake flag-o-matic llvm rocm
+
+GTEST_COMMIT="b85864c64758dec007208e56af933fc3f52044ee"
+GTEST_FILE="gtest-1.14.0_p20220421.tar.gz"
+
+LLVM_MAX_SLOT=17
+
+DESCRIPTION="High Performance Composable Kernel for AMD GPUs"
+HOMEPAGE="https://github.com/ROCmSoftwarePlatform/composable_kernel"
+SRC_URI="https://github.com/ROCmSoftwarePlatform/composable_kernel/archive/rocm-${PV}.tar.gz -> ${P}.tar.gz
+	test? ( https://github.com/google/googletest/archive/${GTEST_COMMIT}.tar.gz -> ${GTEST_FILE} )"
+
+LICENSE="MIT"
+KEYWORDS="~amd64"
+SLOT="0/$(ver_cut 1-2)"
+
+IUSE="debug test"
+REQUIRED_USE="${ROCM_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	dev-util/hip
+	>=dev-db/sqlite-3.17
+	sci-libs/rocBLAS:${SLOT}[${ROCM_USEDEP}]
+	>=dev-libs/boost-1.72
+	dev-cpp/nlohmann_json
+	dev-cpp/frugally-deep
+"
+
+DEPEND="${RDEPEND}"
+
+BDEPEND="dev-libs/half:0/1
+	dev-build/rocm-cmake
+"
+
+S="${WORKDIR}/composable_kernel-rocm-${PV}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-5.7.1-enable-test-examples.patch"
+)
+
+src_configure() {
+	addpredict /dev/kfd
+	addpredict /dev/dri/
+
+	if ! use debug; then
+		append-cflags "-DNDEBUG"
+		append-cxxflags "-DNDEBUG"
+		CMAKE_BUILD_TYPE="Release"
+	else
+		CMAKE_BUILD_TYPE="Debug"
+	fi
+
+	local mycmakeargs=(
+		-DCMAKE_SKIP_RPATH=ON
+		-DAMDGPU_TARGETS="$(get_amdgpu_flags)"
+		-DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr"
+		-DBUILD_TESTS=$(usex test ON OFF)
+	)
+
+	if use test; then
+		mycmakeargs+=(
+			-DGOOGLETEST_DIR="${WORKDIR}/googletest-${GTEST_COMMIT}"
+		)
+	fi
+
+	CXX=hipcc cmake_src_configure
```

Comment:
Its a good idea to strip unsupported flags for hipcc, for example a gcc user may not expect their gcc specific flags to be fed to clang here.

Setting CXX to hipcc and running `strip-unsupported-flags` should handle this.

```
ninja: job failed: /usr/bin/hipcc -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/var/tmp/portage/sci-libs/composable-kernel-5.7.1-r1/work/composable_kernel-rocm-5.7.1/profiler/include -I/var/tmp/portage/sci-libs/composable-kernel-5.7.1-r1/work/composable_kernel-rocm-5.7.1/library/include -I/var/tmp/portage/sci-libs/composable-kernel-5.7.1-r1/work/composable_kernel-rocm-5.7.1/include  -O3 -march=znver2 -pipe -flto=auto -fuse-linker-plugin -frecord-gcc-switches -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -DNDEBUG -std=c++17   -Wall -Wextra -Wcomment -Wendif-labels -Wformat -Winit-self -Wreturn-type -Wsequence-point -Wswitch -Wtrigraphs -Wundef -Wuninitialized -Wunreachable-code -Wunused -Wno-reserved-identifier -Werror -Wsign-compare -Wno-extra-semi-stmt -Wno-missing-field-initializers -Wno-deprecated-declarations -Wall -Wextra -Wcomment -Wendif-labels -Wformat -Winit-self -Wreturn-type -Wsequence-point -Wswitch -Wtrigraphs -Wundef -Wuninitialized -Wunreachable-code -Wunused -Wno-reserved-identifier -Werror -Wsign-compare -Wno-extra-semi-stmt -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-conversion -Wno-double-promotion -Wno-exit-time-destructors -Wno-extra-semi -Wno-float-conversion -Wno-gnu-anonymous-struct -Wno-gnu-zero-variadic-macro-arguments -Wno-missing-prototypes -Wno-nested-anon-types -Wno-padded -Wno-return-std-move-in-c++11 -Wno-shorten-64-to-32 -Wno-sign-conversion -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-weak-vtables -Wno-covered-switch-default -Wno-unsafe-buffer-usage -Werror -Weverything -Wno-global-constructors -x hip --offload-arch=gfx1010 -MD -MT profiler/src/CMakeFiles/ckProfiler.dir/profile_conv_fwd_bias_relu_add.cpp.o -MF profiler/src/CMakeFiles/ckProfiler.dir/profile_conv_fwd_bias_relu_add.cpp.o.d -o profiler/src/CMakeFiles/ckProfiler.dir/profile_conv_fwd_bias_relu_add.cpp.o -c /var/tmp/portage/sci-libs/composable-kernel-5.7.1-r1/work/composable_kernel-rocm-5.7.1/profiler/src/profile_conv_fwd_bias_relu_add.cpp
clang++: error: optimization flag '-fuse-linker-plugin' is not supported [-Werror,-Wignored-optimization-argument]
clang++: error: optimization flag '-fuse-linker-plugin' is not supported [-Werror,-Wignored-optimization-argument]
```

Reply:
> https://github.com/gentoo/gentoo/pull/34869#issuecomment-1954823915

So to clarify a best way to get around this is to have `CXX="clang" strip-unsupported-flags` in the ebuild. it will strip all the clang unsupported flags  that hipcc would feed to it later.